### PR TITLE
Fix rule namespace selector

### PR DESCRIPTION
**Pull request by brancz**
I chose to e2e test here that we are in fact picking up our alerting rules, by testing that the `Watchdog` alert is firing.

@metalmatze @mxinden @s-urbaniak @squat 

cc @josefkarasek

### CODE REVIEW

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -124,6 +124,9 @@ func TestQueryPrometheus(t *testing.T) {
 		}, {
 			query:   `up{job="crio"} == 1`,
 			expectN: 1,
+		}, {
+			query:   `ALERTS{alertname="Watchdog"} == 1`,
```
Comment (mxinden) [inline test/e2e/main_test.go]: This only tests, whether rules from the same namespace are loaded, right?

**Comment by josefkarasek** [comment]
Looks great, thanks!

File: test/e2e/main_test.go (hunk 1 above)
Comment (brancz) [reply test/e2e/main_test.go]: What does "same" namespace mean, when you request "all" namespaces? Sure the object is in the openshift-monitoring namespace, but the Prometheus object doesn't know that at selection time

Comment (brancz) [reply test/e2e/main_test.go]: Added another e2e test that creates a new, non `openshift-monitoring` namespace, that should be picked up from.

File: test/e2e/multi_namespace_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func TestMultinamespacePrometheusRule(t *testing.T) {
+	t.Parallel()
+
+	nsName := "openshift-test-prometheus-rules"
+
+	err := f.OperatorClient.CreateOrUpdateNamespace(&v1.Namespace{
```
Comment (mxinden) [inline test/e2e/multi_namespace_test.go]: Given that the namespace name is not randomized, would you mind also removing it at the end of the tests, making sure two consecutive tests run on personal clusters don't influence each other.

Comment (brancz) [reply test/e2e/multi_namespace_test.go]: https://github.com/openshift/cluster-monitoring-operator/pull/263

**Pull request merged**